rotkehlchen/globaldb/handler.py

```diff
@@ -621,16 +612,11 @@ def add_evm_token_data(write_cursor: DBCursor, entry: EvmToken) -> None:
                 ),
             )
         except sqlite3.IntegrityError as e:
-            exception_msg = str(e)
-            if 'FOREIGN KEY' in exception_msg:
-                # should not really happen since API should check for this
-                msg = (
-                    f'Ethereum token with address {entry.evm_address} can not be put '
-                    f'in the DB due to swapped for entry not existing'
-                )
-            else:
-                msg = f'Ethereum token with address {entry.evm_address} already exists in the DB'  # noqa: E501
-            raise InputError(msg) from e
+            # should not really happen since API should check for this
+            raise InputError(
+                f'Ethereum token with address {entry.evm_address} can not be put '
+                f'in the DB due to {str(e)}',
+            ) from e
```

Comment:
Because the message that was generated here before might be incorrect. And essentially this should not happen, so I thought that it is better to return full error message.